### PR TITLE
update to use current demo standards

### PR DESCRIPTION
**Pull request by eli-darkly**
@hroederld - please see the spec page "SDK Hello World" for the requirements being implemented here.

One code style thing I wasn't sure about: this code uses the type alias `LDBoolean`, whereas the `hello-c-server` code just uses `bool`.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -23,27 +45,35 @@ jobs:
...
           name: Run Hello
-          command: ./hello
+          command: |
+            ./hello | tee output.txt
+            grep "is true for this user" output.txt || (echo "Flag did not evaluate to expected true value" && exit 1)
 
   build-test-osx:
     macos:
-      xcode: "9.0"
+      xcode: "9.4.1"
```
Comment (eli-darkly) [inline .circleci/config.yml:66]: This is a very old Xcode version, but it's what the SDK project is currently building with.

**Comment by hroederld** [comment]
hello-c-server is generally out of date, as to `LDBoolean` it is common for c libraries to define their own boolean types. There are all kinds of projects that do something different for booleans (because of historical compatibility) and the safest route is to define your own type for the public API (the customer can use true / false / bool it is compatible). I really like the CI change to do a real integration test.

**Review by hroederld — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (eli-darkly) [reply .circleci/config.yml:66]: @hroederld Can you comment on whether there's a reason to continue using Xcode 9.x?

Comment (hroederld) [reply .circleci/config.yml:66]: No I don't have a good reason to use an old toolchain for the hello app.

**Pull request merged**